### PR TITLE
Add initial support for code signing services

### PR DESCRIPTION
**Pull request by steven-joruk**
I'm not sure if the GuestAttributes is the right approach, maybe I'm just excessively averse to having to manually create CFDictionaries.

I'm intending to use this in another project to securely validate XPC client connections using their audit token.

### CODE REVIEW

File: security-framework/src/os/macos/code_signing.rs [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+    SecCode, SecCodeRef
+}
+impl_TCFType!(SecCode, SecCodeRef, SecCodeGetTypeID);
+
+impl SecCode {
+    /// Retrieves the code object for the code making the call.
+    pub fn for_self(flags: Flags) -> Self {
+        let mut code = MaybeUninit::uninit();
+
+        let status = unsafe { SecCodeCopySelf(flags.bits(), code.as_mut_ptr()) };
```
Comment (steven-joruk) [inline security-framework/src/os/macos/code_signing.rs:118]: Woops, use cvt and Result

File: security-framework/src/os/macos/code_signing.rs [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+    // - architecture
+    // - canonical
+    // - dynamic code
+    // - dynamic code info plist
+    // - hash
+    // - mach port
+    // - sub-architecture
+
+    /// The guest's audit token.
+    pub fn audit_token(&mut self, token: CFDataRef) {
```
Comment (kornelski) [inline security-framework/src/os/macos/code_signing.rs:65]: I think they should be named `set_*`. Other than that it seems fine. 

Maybe add `set_other<V: ToVoid<V>>(CFStringRef, V)` to allow adding other attributes.

**Comment by steven-joruk** [comment]
I'm not sure what's going on with the stable test failures. I tried using [ctest2](https://github.com/JohnTitor/ctest2) instead which didn't help.

Using `super::` instead of `base::` works around some errors, but it'd require a lot of changes and doesn't solve the underlying issue. I'll keep looking.

**Comment by steven-joruk** [comment]
The lints are failing on rust 1.38 due to a dependency using an unsupported clippy config. Is it okay to bump the rust version for the lints job? Is 1.42 okay?

**Comment by kornelski** [comment]
Yeah, bumping clippy should be fine. Thank you for the PR.

**Pull request merged**